runtime.md

```diff
@@ -15,3 +15,46 @@ Runs a process in a container. Can be invoked several times.
 Not sure we need that from oc cli. Process is killed from the outside.
 
 This event needs to be captured by oc to run onstop event handlers.
+
+## Hooks
+Hooks allow one to run code before/after various lifecycle events of the container. The state of the container is passed to the hooks over stdin, so the hooks could get the information they need to do their work.
+
+The path to the hooks could be absolute or just the name of the file. Hooks are executed from the host's filesystem. If the path to a hook isn't absolute then it is searched for within the PATH environment variable on the host.
```

Comment:
I think this is still missing [“lookup relative to the config-file's directory”][1] (motivation [here][2]).  How about:

> Hooks are executed from the hosts's filesystem and environment.  Hook paths can be:
>
> * an absolute path,
> * a bare command name (without path separators), in which case the command is searched for within the `PATH` environment variable, or
> * a relative path, in which case the command is found relative to the directory containing the config file.  Relative paths cannot reference parent directories (i.e., no `../…`).

That way I can have (with paths like `hook/pre-start.sh`):

    .
    ├── config.json
    ├── hook
    │   ├── post-stop.sh
    │   └── pre-start.sh
    └── rootfs

That way I can have (with paths like `./pre-start.sh`):

    .
    ├── config.json
    ├── post-stop.sh
    ├── pre-start.sh
    └── rootfs

I'm also not sure how any of the non-bare-command cases will be handled in a cross-OS way.  Do we require slash-separators (vs. Windows with it's backslashes)?  How do folks handle Windows drives?  Do we need OS-specific hook sections so folks can use their native path syntax (and possibly different commands / arguments)?

[1]: https://github.com/opencontainers/specs/pull/34#issuecomment-119729756
[2]: https://github.com/opencontainers/specs/pull/34#issuecomment-119714814